documenteer/stackdocs/build.py

```diff
@@ -44,3 +49,128 @@ def discover_setup_packages():
+        Directory of an EUPS package.
+
+    Returns
+    -------
+    doc_dirs : namedtuple
+        Attributes of the named tuple are:
+
+        - ``package_dirs`` (`dict`). Keys are package names (for example,
+          ``'afw'``). Values are absolute directory paths to the package's
+          documentation directory inside the package's ``doc`` directory. If
+          there is no package-level documentation the dictionary will be empty.
+
+        - ``modules_dirs`` (`dict`). Keys are module names (for example,
+          ``'afw.table'``). Values are absolute directory paths to the module's
+          directory inside the package's ``doc`` directory. If a package has no
+          modules the returned dictionary will be empty.
+
+        - ``static_doc_dirs`` (`dict`). Paths of ``_static`` sub-directories.
+          Keys are directory names relative to the ``_static`` directory.
+          Values are absolute directory paths to the static documentation
+          directory in the package. Keys are directo
+
+    Raises
+    ------
+    IOError
+       Raised when the ``manifest.yaml`` file cannot be found in a package.
+
+    Notes
+    -----
+    Stack packages have documentation in subdirectories of their `doc`
+    directory. The ``manifest.yaml`` file declares what these directories are
+    so that they can be symlinked into the root project.
+
+    There are three types of documentation directories:
+
+    1. Package doc directories contain documentation for the EUPS package
+       aspect.
+    2. Module doc directories contain documentation for a Python package
+       aspect.
+    3. Static doc directories are root directories inside the package's
+       ``doc/_static/`` directory.
+
+    These are declared in a package's ``doc/manifest.yaml`` file. For example:
+
+    .. code-block:: yaml
+
+       package: "afw"
+       modules:
+         - "lsst.afw.image"
+         - "lsst.afw.geom"
+       statics:
+         - "_static/afw"
+
+    This YAML declares *module* documentation directories:
+
+    - ``afw/doc/lsst.afw.image/``
+    - ``afw/doc/lsst.afw.geom/``
+
+    It also declares a *package* documentation directory:
+
+    - ``afw/doc/afw``
+
+    And a static documentaton directory:
+
+    - ``afw/doc/_static/afw``
+    """
+    doc_dir = os.path.join(package_dir, 'doc')
+    modules_yaml_path = os.path.join(doc_dir, 'manifest.yaml')
+
+    if not os.path.exists(modules_yaml_path):
+        raise IOError('Manifest YAML not found: {0}'.format(modules_yaml_path))
+
+    with open(modules_yaml_path) as f:
+        manifest_data = yaml.safe_load(f)
+
+    module_dirs = {}
+    package_dirs = {}
+    static_dirs = {}
+
+    if 'modules' in manifest_data:
+        for module_name in manifest_data['modules']:
+            module_dir = os.path.join(doc_dir, module_name)
+
+            # validate that the module's documentation directory does exist
+            if not os.path.isdir(module_dir):
+                continue
```

Comment:
I'm a little confused by this.  If a module is in the manifest but the doc directory doesn't exist, shouldn't it complain a little louder than silently skipping?

Ignore this.  It is taken care of in a later commit.